capellambse/model/crosslayer/information.py

```diff
@@ -46,38 +66,175 @@ def _search_all_exchanges(
+    """A Property of a Class."""
+
+    _xmltag = "ownedFeatures"
+
+    is_ordered = xmltools.BooleanAttributeProperty(
+        "_element",
+        "ordered",
+        __doc__="Boolean flag, indicates if property is ordered",
+    )
+    is_unique = xmltools.BooleanAttributeProperty(
+        "_element",
+        "unique",
+        __doc__="Boolean flag, indicates if property is unique",
+    )
+    is_abstract = xmltools.BooleanAttributeProperty(
+        "_element",
+        "isAbstract",
+        __doc__="Boolean flag, indicates if property is abstract",
+    )
+    is_static = xmltools.BooleanAttributeProperty(
+        "_element",
+        "isStatic",
+        __doc__="Boolean flag, indicates if property is static",
+    )
+    is_part_of_key = xmltools.BooleanAttributeProperty(
+        "_element",
+        "isPartOfKey",
+        __doc__="Boolean flag, indicates if property is part of key",
+    )
+    is_derived = xmltools.BooleanAttributeProperty(
+        "_element",
+        "isDerived",
+        __doc__="Boolean flag, indicates if property is abstract",
+    )
+    is_read_only = xmltools.BooleanAttributeProperty(
+        "_element",
+        "isReadOnly",
+        __doc__="Boolean flag, indicates if property is read-only",
+    )
+    visibility = xmltools.EnumAttributeProperty(
+        "_element", "visibility", modeltypes.VisibilityKind, default="UNSET"
+    )
+    type = c.AttrProxyAccessor(c.GenericElement, "abstractType")
+    default_value = property(create_role_tag_getter("ownedDefaultValue"))
+    min = property(create_role_tag_getter("ownedMinValue"))
+    max = property(create_role_tag_getter("ownedMaxValue"))
+    null_value = property(create_role_tag_getter("ownedNullValue"))
+    min_card = property(create_role_tag_getter("ownedMinCard"))
+    max_card = property(create_role_tag_getter("ownedMaxCard"))
+
+
 @c.xtype_handler(None)
 class Class(c.GenericElement):
     """A Class."""
 
     _xmltag = "ownedClasses"
 
     sub = c.Accessor
-    super = c.Accessor
+    super: Class = c.Accessor  # type: ignore
+    is_abstract = xmltools.BooleanAttributeProperty(
+        "_element",
+        "abstract",
+        __doc__="Boolean flag, indicates if class is abstract",
+    )
+    is_final = xmltools.BooleanAttributeProperty(
+        "_element",
+        "final",
+        __doc__="Boolean flag, indicates if class is final",
+    )
+    is_primitive = xmltools.BooleanAttributeProperty(
+        "_element",
+        "isPrimitive",
+        __doc__="Boolean flag, indicates if class is primitive",
+    )
+    visibility = xmltools.EnumAttributeProperty(
+        "_element", "visibility", modeltypes.VisibilityKind, default="UNSET"
+    )
     state_machines = c.ProxyAccessor(
         capellacommon.StateMachine, aslist=c.ElementList
     )
+    own_properties = c.ProxyAccessor(
+        Property, aslist=c.ElementList, follow_abstract=False
+    )
+
+    @property
+    def properties(self) -> c.ElementList[Property]:
```

Comment:
from the class perspective though - it has properties which could be own but also inherited. So I'd rather keep it as-is in this case. When we want to alter a list of class properties we would always do that on own_ but when we want to see the list of class properties we'd want to see the complete list including the inherited. I the end-user role we may not know that class has some properties inherited from super so having those listed via default "properties" makes it more safe against missing things out. Can we keep it then, @Wuestengecko ?

Reply:
That makes sense. For consistency with other parts of the model, we'd probably want to use `owned_properties` instead of `own_properties` though (see e.g. `PhysicalComponent.owned_components`).

Reply:
agree, lets rename to `owned_properties`